### PR TITLE
chore(headless)!: remove headless fetch & crypto polyfills

### PR DESCRIPTION
**Pull request by alexprudhomme**
https://coveord.atlassian.net/browse/KIT-3023

This PR is part of the making sure there are no 'require's in the headless ESM output.
#### Before

image

#### After

image

-  [x] Remove undici from renovate.json

There is still this one that we will have to change or make it only available to the commonJs output. https://github.com/coveo/ui-kit/blob/KIT-3023/packages/headless/ponyfills/abortable-fetch-shim.js

### CODE REVIEW

**Comment by louis-bompart** [comment]
@coveo/salesforce-integration  can you help @alexprudhomme  check/ensure that the IIFE bundle for Quantic are still kosher?

**Comment by alexprudhomme** [comment]
> @coveo/salesforce-integration can you help @alexprudhomme check/ensure that the IIFE bundle for Quantic are still kosher?

I think it should be fine since I did not touch the quanticUmd build phase in headless/dist/quantic

https://github.com/coveo/ui-kit/blob/KIT-3023/packages/headless/esbuild.mjs#L140-L163

**Review by olamothe — APPROVED** [review]

**Comment by mmitiche** [comment]
> @coveo/salesforce-integration can you help @alexprudhomme check/ensure that the IIFE bundle for Quantic are still kosher?

I will do a quick sanity check/test from this PR to double check the Headless quantic bundle is still working property inside Salesforce.

**Comment by erocheleau** [comment]
Yea we need to check with multiple configuration after this has changed:

1. With LWS enabled
2. With LockerService instead of LWS
3. With Locker disabled in a community.
4. In a LWR site.

@mmitiche if you have the time, please put ✅ if you have tested some of these, I can try to test the others :)

File: packages/headless/esbuild.mjs [hunk 1]
```diff
@@ -190,10 +190,6 @@ async function buildBrowserConfig(options, outDir) {
     plugins: [
       alias({
         'coveo.analytics': resolveEsm('coveo.analytics'),
-        '@coveo/please-give-me-fetch': resolve(
-          './ponyfills',
-          'fetch-ponyfill-browser.js'
```
Comment (erocheleau) [inline packages/headless/esbuild.mjs:195]: In think until we figure out why we needed to polyfill fetch for a browser, we shouldn't be confident to remove it?

**Review by erocheleau — CHANGES_REQUESTED** [review]
Just making sure this isn't merged before it has been properly tested

**Comment by mmitiche** [comment]
Quantic working properly with:

- With LWS enabled ✅
- With LockerService instead of LWS ✅
- With Locker disabled in a community ✅ 
- In a LWR site ✅

**Review by fbeaudoincoveo — APPROVED** [review]

**Review by erocheleau — APPROVED** [review]
Looks good, thanks for making the tests!

I also confirmed that none of the Crypto stuff was needed due to any Locker/LWS limitation or distortion!

GG

**Pull request merged**